editor/plugins/script_text_editor.h

```diff
@@ -196,6 +196,7 @@ class ScriptTextEditor : public ScriptEditorBase {
 	void drop_data_fw(const Point2 &p_point, const Variant &p_data, Control *p_from);
 
 	String _get_absolute_path(const String &rel_path);
+	static String _quote_drop_data(const String &str);
```

Comment:
No need to put it in the header if it's static.

Reply:
I'm trying to figure out what the best way to define this type of method:
- It's a helper method for ScriptTextEditor
- It's not used outside of ScriptTextEditor (therefore shouldn't be in the .h file like you said)
- Someone in dev chat was saying it's uncommon to have methods that aren't associated with a class.

`String _quote_drop_data(const String &str) { ... }` seems to counter that last point.
`static String ScriptTextEditor::_quote_drop_data(const String &str) { ... }` doesn't compile without it also in the .h file
`static String _quote_drop_data(const String &str) { ... }` compiles but not entirely sure what it does, hehe

Haven't done C++ in almost 2 decades, so probably missing something here :)

Reply:
Pushed a change using the first method, which I think was your suggestion. Happy to change if it wasn't :)

Reply:
Should be the latter one `static String _quote_drop_data(const String &str) { ... }` :)

It's from C to make the function only visible in this file.

It's not the most common pattern, but used a few times in the `ScriptEditors` for small utility methods like this, see `_find_script_node` above this one. There's a few more further up too.

Reply:
Changes submitted!

I was assuming C++, will look into static in C, thank you!